gammapy/scripts/image_pipe.py

```diff
@@ -2,56 +2,196 @@
 from __future__ import (absolute_import, division, print_function,
                         unicode_literals)
 import logging
+import numpy as np
+from astropy.coordinates import Angle
+from astropy.units import Quantity
+from astropy.io import fits
 from ..utils.scripts import get_parser
+from ..background import fill_acceptance_image
+from ..image import SkyMap, SkyMapCollection, disk_correlate
+from ..stats import significance
 
 __all__ = ['ImageAnalysis']
 
 log = logging.getLogger(__name__)
 
 
-def image_pipe_main(args=None):
-    parser = get_parser(ImageAnalysis)
-    parser.add_argument('config_file', type=str,
-                        help='Config file in YAML format')
-    # TODO: add option to dump the default config file
-    args = parser.parse_args(args)
-    analysis = ImageAnalysis.from_yaml(args.config_file)
-    analysis.run()
-
-
 class ImageAnalysis(object):
     """Gammapy 2D image based analysis.
+
+    Parameters
+    ----------
+    nxpix : int, optional
+            Number of pixels in x axis. Default is 200.
+    nypix : int, optional
+            Number of pixels in y axis. Default is 200.
+    binsz : float, optional
+            Bin size for x and y axes in units of degrees. Default is 0.02.
+    xref : float, optional
+            Coordinate system value at reference pixel for x axis.
+    yref : float, optional
+            Coordinate system value at reference pixel for y axis.
+    proj : string, optional
+            Any valid WCS projection type. Default is 'CAR' (cartesian).
+    coordsys : {'CEL', 'GAL'}, optional
+            Coordinate system, either Galactic ('GAL') or Equatorial ('CEL').
+    energy_band : `~gammapy.utils.energy.Energy
+        Energy band for which we want to compute the image
+    offset_band :`astropy.coordinates.Angle`
+        Offset Band where you compute the image
+    data_store : `~gammapy.data.DataStore`
+        `DataStore` where are situated the events
+    counts_image : ~gammapy.image.SkyMap`
+            count image
+    bkg_image : ~gammapy.image.SkyMap`
+            bkg image
     """
 
-    def __init__(self, config):
-        self.config = config
-
-    @classmethod
-    def from_yaml(cls, filename):
-        """Read config from YAML file."""
-        import yaml
-        log.info('Reading {}'.format(filename))
-        with open(filename) as fh:
-            config = yaml.safe_load(fh)
-        return cls(config)
-
-    def run(self):
-        """Run analysis chain."""
-        log.info('Running analysis ...')
-        print(self.config['general']['outdir'])
-        print(self.config)
-
-    def make_counts(self):
+    def __init__(self, nxpix=None, nypix=None, binsz=None, xref=None, yref=None, proj=None, coordsys=None,
+                 energy_band=None, offset_band=None,
+                 data_store=None, counts_image=None, bkg_image=None):
+        self.maps = SkyMapCollection()
```

Comment:
Yes, I think using the `SkyMapCollection` is much more flexible and convenient. Looks much better to me!